src/schnorr.rs

```diff
@@ -25,9 +25,14 @@ pub struct ID {
 #[allow(non_snake_case)]
 impl ID {
     /// Construct a new schnorr ID which binds the passed `Scalar` `id` and `Scalar` `a`, with a zero-knowledge proof of ownership of `a`
-    pub fn new<RNG: RngCore + CryptoRng>(id: &Scalar, a: &Scalar, rng: &mut RNG) -> Self {
+    pub fn new<RNG: RngCore + CryptoRng>(
+        id: &Scalar,
+        a: &Scalar,
+        ctx: &[u8],
```

Comment:
We should update the docstrings here to describe what this is expected to be. Also, is the expectation here that ctx should be 8 bytes, 32 bytes, or whatever the user wants is fine?

Reply:
Oh this can be used to make sure that the length of the hashed bytes is the same, so maybe this can be whatever the user wants.